metrics/lib/common.bash

```diff
@@ -284,48 +284,50 @@ check_for_ksm(){
 #
 # arg1 - timeout in seconds
 wait_ksm_settle(){
-	local t pcnt
-	local oldscan=-1 newscan
-	local oldpages=-1 newpages
-
-	oldscan=$(cat /sys/kernel/mm/ksm/full_scans)
-
-	# Go around the loop until either we see a small % change
-	# between two full_scans, or we timeout
-	for ((t=0; t<$1; t++)); do
-
-		newscan=$(cat /sys/kernel/mm/ksm/full_scans)
-		if (( newscan != oldscan )); then
-			echo -e "\nnew full_scan ($oldscan to $newscan)"
-
-			newpages=$(cat /sys/kernel/mm/ksm/pages_shared)
-			# Do we have a previous scan to compare with
-			echo "check pages $oldpages to $newpages"
-			if (( oldpages != -1 )); then
-				# avoid divide by zero problems
-				if (( $oldpages > 0 )); then
-					pcnt=$(( 100 - ((newpages * 100) / oldpages) ))
-					# abs()
-					pcnt=$(( $pcnt * -1 ))
-
-					echo "$oldpages to $newpages is ${pcnt}%"
-
-					if (( $pcnt <= 5 )); then
-						echo "KSM stabilised at ${t}s"
-						return
+	if [ "$RUNTIME" = "kata-runtime" ] || [ "$RUNTIME" == "kata-qemu" ]; then
+		local t pcnt
+		local oldscan=-1 newscan
+		local oldpages=-1 newpages
+
+		oldscan=$(cat /sys/kernel/mm/ksm/full_scans)
+
+		# Go around the loop until either we see a small % change
+		# between two full_scans, or we timeout
+		for ((t=0; t<$1; t++)); do
+
+			newscan=$(cat /sys/kernel/mm/ksm/full_scans)
+			if (( newscan != oldscan )); then
+				echo -e "\nnew full_scan ($oldscan to $newscan)"
+
+				newpages=$(cat /sys/kernel/mm/ksm/pages_shared)
+				# Do we have a previous scan to compare with
+				echo "check pages $oldpages to $newpages"
+				if (( oldpages != -1 )); then
+					# avoid divide by zero problems
```

Comment:
Somewhere around here, I think we maybe can code up that 'we know we have done a complete ksm scan', and that if there have been no pages found to merge yet, then we could probably fairly safely assume the test is not utilising KSM, and thus can quit early.
wdyt @GabyCT - I think it will take a little bit of concentration to ensure we get the right spot here to do that check...